### PR TITLE
feat: [CDS-95808]: design system review and dark mode support

### PR DESCRIPTION
**Pull request by abhinavrastogi-harness**
| After | Before |
|--------|--------|
| Screenshot 2024-04-18 at 3 07 17 PM | Screenshot 2024-04-18 at 3 07 28 PM |
| Screenshot 2024-04-18 at 3 09 06 PM | Screenshot 2024-04-18 at 3 09 13 PM |

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- StyleLint: `retrigger stylelint`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- ImageSnapshot `retrigger ImageSnapshot`

### CODE REVIEW

**Comment by bot2-harness** [comment]
Image snapshots failed please check https://console.cloud.google.com/storage/browser/ui-team-bucket/uicore-2372

**Comment by bot2-harness** [comment]
Image snapshots failed please check https://console.cloud.google.com/storage/browser/ui-team-bucket/uicore-2399

File: packages/uicore/src/styles/colors.css
Comment (abhinavrastogi-harness) [inline packages/uicore/src/styles/colors.css:1]: slight change in light-mode colors to match the design system, as provided by Joanna on the UX team.

**Comment by bot2-harness** [comment]
Image snapshots failed please check https://console.cloud.google.com/storage/browser/ui-team-bucket/uicore-2441

**Comment by bot2-harness** [comment]
Image snapshots failed please check https://console.cloud.google.com/storage/browser/ui-team-bucket/uicore-2442

**Review by sahithikolichala — APPROVED** [review]

**Review by vardanbansal-harness — APPROVED** [review]

**Review by pankajsingh-harness — APPROVED** [review]

**Review by mayankVermaHarness — APPROVED** [review]

**Pull request merged**